documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+designed to easily create common configuration types such as direct MQTT or WebSocket connections. After an MQTT5 client
+is built and finalized, the settings of the resulting MQTT5 client cannot be modified.
+
+#### Example of creating a client in the v1 SDK
+
+```typescript
+var awsIot = require('aws-iot-device-sdk');
+
+var clientEndpoint = "<prefix>-ats.iot.<region>.amazonaws.com";
+var clientId = "<unique client id>";
+var certificateFile = "<certificate file>";  // X.509 based certificate file
+var privateKeyFile = "<private key file>";   // PEM encoded private key file
+
+var device = awsIot.device({
+        keyPath: privateKeyFile,
+        certPath: certificateFile,
+        clientId: clientId,
+        host: clientEndpoint
+});
+```
+
+#### Example of creating a client in the v2 SDK
+
+The v2 SDK supports different connection types. Given the same input parameters as in the v1 example above, the recommended
+method to create an MQTT5 client will be [newDirectMqttBuilderWithMtlsFromPath](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/iot.AwsIotMqtt5ClientConfigBuilder.html#newDirectMqttBuilderWithMtlsFromPath). 
+
+```typescript
+import { mqtt5, iot } from "aws-iot-device-sdk-v2";
+
+const clientEndpoint : string = "<prefix>-ats.iot.<region>.amazonaws.com";
+const clientId : string = "<unique client id>";
+const certificateFile : string = "<certificate file>";  // X.509 based certificate file
+const privateKeyFile : string = "<private key file>";   // PEM encoded private key file
+
+let builder = iot.AwsIotMqtt5ClientConfigBuilder.newDirectMqttBuilderWithMtlsFromPath(
+        clientEndpoint,
+        certificateFile,
+        privateKeyFile
+
+builder.withConnectProperties({
+    clientId: clientId
+});
+
+let config : mqtt5.Mqtt5ClientConfig = builder.build();
+
+let client : mqtt5.Mqtt5Client = new mqtt5.Mqtt5Client(config);
+```
+
+For more information, refer to the [Connection types and features](#connection-types-and-features) section for other connection types supported by the v2 SDK.
+
+### Client start
+
+To connect to the server in the v1 SDK, you just instantiate a `device` object.
+
+In the v2 SDK, you explicitly `start` the MQTT5 client.
+
+#### Example of connecting to a server in the v2 SDK
+
+```typescript
+let config : mqtt5.Mqtt5ClientConfig = builder.build();
+let client : mqtt5.Mqtt5Client = new mqtt5.Mqtt5Client(config);
+client.start();
+```
+
+### Connection types and features
+
+The v1 SDK supports four types of connections to the AWS IoT service: MQTT with X.509 certificate, MQTT over Secure
+WebSocket with SigV4 authentication, MQTT over Secure WebSocket with Cognito authentication, and MQTT over Secure
+WebSocket using a custom authorization function to authenticate.
+
+The v2 SDK adds a collection of connection types and cryptography formats (e.g. [PKCS #11](https://en.wikipedia.org/wiki/PKCS_11)),
```

Comment:
Maybe we should add these additional connection types to the "What's New" section up top. They seem like a material addition to the capabilities of v2 over v1.